x/oracle/types/ballot.go

```diff
@@ -132,6 +133,27 @@ func (pb ExchangeRateBallot) Swap(i, j int) {
 	pb[i], pb[j] = pb[j], pb[i]
 }
 
+// BallotDenom is a convenience wrapper for setting rates deterministically.
+type BallotDenom struct {
+	Ballot ExchangeRateBallot
+	Denom  string
+}
+
+// BallotMapToSlice returns an array of sorted exchange rate ballots.
+func BallotMapToSlice(votes map[string]ExchangeRateBallot) []BallotDenom {
+	b := make([]BallotDenom, len(votes))
+	for denom, ballot := range votes {
+		b = append(b, BallotDenom{
```

Comment:
So we have to use a index variable here since we've pre-allocated the slice.
```suggestion
		b[i] = BallotDenom{
```